Assets/Scripts/RagdollController.cs

```diff
@@ -53,8 +62,35 @@ public void EnableRagdoll(Vector3 knockbackForce)
         rigidbodyToPush.AddForce(knockbackForce, ForceMode.Impulse);
     }
 
-    public void ReparentCamera(Transform cameraTransform)
+    public void ReparentCamera(Camera camera, Transform input)
+    {
+        //camera.transform.parent = cameraParent;
+        camera.cullingMask = ragdollMask;
+        this.camera = camera.transform;
+        inputTransform = input;
+        StartCoroutine(WaitAndDisableCamera(camera));
+    }
+
+    private IEnumerator WaitAndDisableCamera(Camera camera)
+    {
+        yield return new WaitForSeconds(7f);
+        camera.enabled = false;
+    }
+
+    private void Update()
+    {
+        if (!camera)
+            return;
+        camera.position = cameraParent.position + new Vector3(0, 2, -5f);
+        camera.LookAt(cameraParent);
+    }
+
+    private void OnDestroy()
     {
-        cameraTransform.parent = cameraParent;
+        if(!camera)
+            return;
+        camera.position = Vector3.zero;
+        camera.localPosition = Vector3.zero;
+        camera.rotation = inputTransform.rotation;
```

Comment:
This doesn't reset the culling mask (not really that necessary I suppose)

Reply:
Ye, it gets automatically changed back on playerManager start